pkg/controller/manager.go

```diff
@@ -52,6 +52,15 @@ func (m *Manager) UpdateController(name string, params ControllerParams) {
 	m.updateController(name, params)
 }
 
+func delayUpdate(ctrl *managedController, params ControllerParams) {
+	select {
+	case <-time.After(params.DelayUpdate):
+		ctrl.update <- params
+	case <-ctrl.terminated:
```

Comment:
nit: The first line here could block, should there be another `select {}` statement to handle the `<-ctrl.terminated` case after the delay?

Reply:
Consider this case: `After()` was called and the goroutine made no progress for some undefined amount of time before proceeding to execute the `ctrl.update <- params` line. 
The `runController()` code that receives from that channel could have received `<-c.stop` immediately before the `After()` returns, then `runController()` proceeds to call into `params.StopFunc(...)` but not yet called `close(c.terminated)`. In this case, this goroutine would be stuck on `ctrl.update <- params` forever because the goroutine which normally reads from that channel will never read from it again.